src/stream/ascii_reader.rs

```diff
@@ -0,0 +1,371 @@
+
+/// Ascii property lists are used in legacy settings and only support four
+/// datatypes: Array, Dictionary, String and Data.
+/// See [Apple
+/// Documentation](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/PropertyLists/OldStylePlists/OldStylePLists.htf
+/// for more infos.
+/// However this reader also support Integers as first class datatype.
+/// This reader will accept certain ill-formed ascii plist without complaining.
+/// It does not check the integrity of the plist format.
+use std::{
+    io::{Read, Seek, SeekFrom},
+};
+use crate::{
+    error::{Error, ErrorKind},
+    stream::Event,
+    Integer,
+};
+
+pub struct AsciiReader<R: Read + Seek> {
+    reader: R,
+    current_pos: u64,
+}
+
+impl<R: Read + Seek> AsciiReader<R> {
+    pub fn new(reader: R) -> Self {
+        Self {
+            reader,
+            current_pos: 0,
+        }
+    }
+
+    fn error(&self, kind: ErrorKind) -> Error {
+        kind.with_byte_offset(self.current_pos)
+    }
+
+    /// Get a char without consuming it.
+    fn peek(&mut self) -> Option<u8> {
+        // consume a char then rollback to previous position.
```

Comment:
It would be nice to avoid the `Seek` bound on the reader.

How about storing a `peeked_byte: Option<u8>` in `AsciiReader` in `peek` and consuming it if it exists in `advance`?

Reply:
Oh you're right. That should be pretty simple to add and will indeed remove the `Seek` requirement.